### ISSUE TITLE
Find vectors furthest away / most distant from query

### ISSUE THREAD

**Issue by MalteEbner** [facebookresearch/faiss#1871]
# Summary
I would like to find the vector furthest away from the query vector. How can I do that? I tried `index.search(query, k=-1)`, but it fails as k must be positive.

```python
dimension = 16
query_vectors = np.random.random((5, dimension)).astype('f')
data = np.random.random((100, dimension)).astype('f')

# how to do it without faiss
distance_matrix = sklearn.metrics.pairwise_distances(query_vectors, data)
indizes_most_distant = np.argmax(distance_matrix, axis=1)

# expected with faiss
faiss_index = faiss.IndexFlatL2(dimension)
faiss_index.add(data)
indizes_most_distant = faiss_index.search(query_vectors, k=-1)[1][:, 0]
```

**Comment by Margie524** [comment]
Why is it that I am involved in your guys app in general when I've never
had anything to do with your app


(github) I've never had to use your app for anything so why do I have to
use it now

**Comment by Margie524** [comment]
You guys got into my information and did what you did. I explain myself
plenty of times that it wasn't me. Not going to click on links that I do
not know what they do

**Comment by Margie524** [comment]
Now you guys put it where I can't even apply back to you I have to do it
like this see how sketched out that looks.

**Comment by Margie524** [comment]
What are you guys doing to my stuff!!!
Plz give me access to my own profile back

**Comment by Margie524** [comment]
At this point it just sounds like always I'm talking to robot

**Comment by Margie524** [comment]
I feel violated even more Because you guys are not listening to me. I don't
understand your web links.

**Comment by Margie524** [comment]
I'm actually just trying to get you guys's website away from my emails I'm not able to log into my Facebook and I don't know why and I feel like your guys's website has something to do with it you guys put a code or a link into my Facebook database or something like that but I just need you guys to help me fix it and get it removed because I've never had to deal with your guys's website ever my entire life and now it's becoming a problem

**Comment by mdouze** [comment]
This is not implemented in Faiss. 
As a special case, if you have normalized vectors, it is possible to find the most distant one by querying the negated vector.
